test/unit/math/rev/mat/prob/ordered_logistic_test.cpp

```diff
@@ -0,0 +1,254 @@
+
+  std::vector<int> y{1, 2, 3, 4};
+
+  vector_v lam_v(4);
+  lam_v << -1.32, -0.05, 0.56, 1.55;
+
+  vector_d lam_d(4);
+  lam_d << -1.32, -0.05, 0.56, 1.55;
+
+  vector_v c_v(3);
+  c_v << -0.95, -0.10, 0.95;
+
+  vector_d c_d(3);
+  c_d << -0.95, -0.10, 0.95;
+
+  AVAR out_v = ordered_logistic_lpmf(y, lam_v, c_d);
+  out_v.grad();
+
+  EXPECT_FLOAT_EQ(out_v.val(), -3.9442226297351447171906);
+
+  EXPECT_FLOAT_EQ(lam_v[0].adj(), -0.40854102156722);
+  EXPECT_FLOAT_EQ(lam_v[1].adj(), -0.223446899109214);
+  EXPECT_FLOAT_EQ(lam_v[2].adj(), -0.062977689154598);
+  EXPECT_FLOAT_EQ(lam_v[3].adj(), 0.354343693774205);
+
+  out_v = ordered_logistic_lpmf(y, lam_d, c_v);
+  out_v.grad();
+
+  EXPECT_FLOAT_EQ(out_v.val(), -3.9442226297351447171906);
+
+  EXPECT_FLOAT_EQ(c_v[0].adj(), -0.62697485850346);
+  EXPECT_FLOAT_EQ(c_v[1].adj(), 0.37990895444219);
+  EXPECT_FLOAT_EQ(c_v[2].adj(), 0.587687820118095);
+}
+
+TEST(ProbDistributionsOrdLog, vv_stvec) {
+  using stan::math::ordered_logistic_lpmf;
+  using stan::math::var;
+  using stan::math::vector_d;
+  using stan::math::vector_v;
+
+  std::vector<int> y{1, 2, 3, 4};
+
+  vector_v lam_v(4);
+  lam_v << -1.32, -0.05, 0.56, 1.55;
+
+  vector_v c1_v(3);
+  c1_v << -0.95, -0.10, 0.95;
+
+  vector_v c2_v(3);
+  c2_v << -0.95, -0.10, 0.95;
+
+  vector_v c3_v(3);
+  c3_v << -0.95, -0.10, 0.95;
+
+  vector_v c4_v(3);
+  c4_v << -0.95, -0.10, 0.95;
+
+  std::vector<vector_v> std_c_v{c1_v, c2_v, c3_v, c4_v};
+
+  AVAR out_v = ordered_logistic_lpmf(y, lam_v, std_c_v);
+  out_v.grad();
+
+  EXPECT_FLOAT_EQ(out_v.val(), -3.9442226297351447171906);
+
+  EXPECT_FLOAT_EQ(lam_v[0].adj(), -0.40854102156722);
+  EXPECT_FLOAT_EQ(lam_v[1].adj(), -0.223446899109214);
+  EXPECT_FLOAT_EQ(lam_v[2].adj(), -0.062977689154598);
+  EXPECT_FLOAT_EQ(lam_v[3].adj(), 0.354343693774205);
+
+  EXPECT_FLOAT_EQ(std_c_v[0][0].adj(), 0.40854102156722);
+  EXPECT_FLOAT_EQ(std_c_v[0][1].adj(), 0.0);
+  EXPECT_FLOAT_EQ(std_c_v[0][2].adj(), 0.0);
+
+  EXPECT_FLOAT_EQ(std_c_v[1][0].adj(), -1.03551588007068);
+  EXPECT_FLOAT_EQ(std_c_v[1][1].adj(), 1.25896277917989);
+  EXPECT_FLOAT_EQ(std_c_v[1][2].adj(), 0.0);
+
+  EXPECT_FLOAT_EQ(std_c_v[2][0].adj(), 0.0);
+  EXPECT_FLOAT_EQ(std_c_v[2][1].adj(), -0.879053824737702);
+  EXPECT_FLOAT_EQ(std_c_v[2][2].adj(), 0.942031513892299);
+
+  EXPECT_FLOAT_EQ(std_c_v[3][0].adj(), 0.0);
+  EXPECT_FLOAT_EQ(std_c_v[3][1].adj(), 0.0);
+  EXPECT_FLOAT_EQ(std_c_v[3][2].adj(), -0.354343693774205);
+}
+
+TEST(ProbDistributionsOrdLog, vd_stvec) {
+  using stan::math::ordered_logistic_lpmf;
+  using stan::math::var;
+  using stan::math::vector_d;
+  using stan::math::vector_v;
+
+  std::vector<int> y{1, 2, 3, 4};
+
+  vector_v lam_v(4);
+  lam_v << -1.32, -0.05, 0.56, 1.55;
+
+  vector_v c1_v(3);
+  c1_v << -0.95, -0.10, 0.95;
+
+  vector_v c2_v(3);
+  c2_v << -0.95, -0.10, 0.95;
+
+  vector_v c3_v(3);
+  c3_v << -0.95, -0.10, 0.95;
+
+  vector_v c4_v(3);
+  c4_v << -0.95, -0.10, 0.95;
```

Comment:
These tests are telling us the gradients are getting to the right places. Any chance you could change a couple of these c vectors though? A code that mistakingly only looked at the first cutpoint vector would pass the test. Doesn't have to be much.